tests/test_mace_fitting.py

```diff
@@ -0,0 +1,56 @@
+import yaml, os
+import time
+from pathlib import Path
+import mace, json
+from wfl.fit.mace import fit
+from wfl.configset import ConfigSet
+import ase.io
+
+# This part is used to detect the location of 'run_train.py' file. 
+mace_dist_dir = list(Path(mace.__file__).parent.parent.glob("mace-*"))[0]
+jsonfile = str(Path(mace_dist_dir / "direct_url.json"))
+with open(jsonfile) as f:
+    d = json.load(f)
+script_dir = d["url"].split(":")[-1][2:]
+mace_fit_cmd = f"python {script_dir}/scripts/run_train.py"
+
+
+def test_mace_fit_from_list(request, tmp_path, monkeypatch, mace_fit_cmd=mace_fit_cmd):
+
+    monkeypatch.chdir(tmp_path)
+    print('getting fitting data from ', request.fspath)
+    print("here :  ", Path(request.fspath))
+
+    params_file_path = Path(Path(request.fspath).parents[0] / 'assets' / 'mace_fit_parameters.yaml')
```

Comment:
Please remove this and all the other unnecessary `Path(Path(...)` calls.   The `/` operator returns a `Path` object, and `tmp_path` is guaranteed to be a `Path`, so you don't have to re-convert it repeatedly.

Also, you can just define
```python
parent_path = Path(request.fspath).parent
```
and use that everywhere in the function.

Reply:
Also, `request.fspath` has been deprected https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path, although I haven't figured out the approved replacement yet.

Reply:
https://docs.pytest.org/en/7.4.x/reference/reference.html#std-fixture-request
Looks like it's `request.path`, which is already of type `Path`.